Can't compile out of the box with several CPUs

There is an issue with the current makefile, because the second CPU tries to process `mkdir obj/extern` while the first one is busy with `mkdir obj`.

The issue only appears in clean clones of the repository though, since `make clean` does not remove the obj and obj/extern directories.

```
$ make -j4
mkdir obj
mkdir obj/extern
g++ -x c++-header stdafx.h -MMD -Werror -Wall -std=c++0x -Wno-sign-compare -Wno-unused-variable -Wfatal-errors -Wno-shift-count-overflow -Wno-tautological-constant-out-of-range-compare -I. -I./extern -g -o stdafx.h.gch
mkdir: cannot create directory 'obj/extern': No such file or directory
Makefile:99: recipe for target 'obj/extern' failed
make: *** [obj/extern] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -92,12 +92,7 @@ DEPS = $(addprefix $(OBJDIR)/,$(SRCS:.cpp=.d))
 ##############################################################################
 
 
-all: $(OBJDIR) $(OBJDIR)/extern $(NAME)
-
-$(OBJDIR):
-	mkdir $(OBJDIR)
-$(OBJDIR)/extern:
-	mkdir $(OBJDIR)/extern
+all: $(NAME)
 
 stdafx.h.gch: stdafx.h
 	$(CC) -x c++-header $< -MMD $(CFLAGS) -o $@

--- a/obj-opt/.gitignore
+++ b/obj-opt/.gitignore
@@ -1,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore

--- a/obj-opt/extern/.gitignore
+++ b/obj-opt/extern/.gitignore
@@ -1,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore

--- a/obj/.gitignore
+++ b/obj/.gitignore
@@ -1,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore

--- a/obj/extern/.gitignore
+++ b/obj/extern/.gitignore
@@ -1,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore

